Frameworks/Foundation/NSDecimalNumber.mm

```diff
@@ -14,296 +14,344 @@
+    return [self initWithUnsignedLongLong:static_cast<unsigned long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-+ (NSDecimalNumber*)one {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithUnsignedInt:(unsigned int)value {
+    return [self initWithUnsignedLongLong:static_cast<unsigned long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-+ (NSDecimalNumber*)zero {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithUnsignedInteger:(NSUInteger)value {
+    return [self initWithUnsignedLongLong:static_cast<unsigned long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-+ (NSDecimalNumber*)notANumber {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithUnsignedLong:(unsigned long)value {
+    return [self initWithUnsignedLongLong:static_cast<unsigned long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithDecimal:(NSDecimal)decimal {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithUnsignedShort:(unsigned short)value {
+    return [self initWithUnsignedLongLong:static_cast<unsigned long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithMantissa:(unsigned long long)mantissa exponent:(short)exponent isNegative:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithUnsignedLongLong:(unsigned long long)value {
+    return [self initWithMantissa:value exponent:0 isNegative:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithString:(NSString*)numericString {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithLongLong:(long long)value {
+    long long val = value;
+    if (val < 0) {
+        val *= -1;
+    }
+    return [self initWithMantissa:val exponent:0 isNegative:(value < 0)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithString:(NSString*)numericString locale:(id)locale {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (instancetype)initWithDecimal:(NSDecimal)decimal {
+    if (self = [super init]) {
+        _decimal = {};
+        NSDecimalCopy(&_decimal, &decimal);
+        NSDecimalCompact(&_decimal);
+    }
+
+    return self;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (NSDecimalNumber*)decimalNumberByAdding:(NSDecimalNumber*)decimalNumber {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (instancetype)initWithMantissa:(unsigned long long)mantissa exponent:(short)exponent isNegative:(BOOL)flag {
+    NSDecimal decimal = {};
+    decimal._exponent = exponent;
+    decimal._isNegative = flag;
+
+    // copy over the 8 byte value.
+    memcpy(&(decimal._mantissa), &mantissa, sizeof(unsigned long long));
+
+    for (int index = NSDecimalMaxSize - 1; index >= 0; --index) {
```

Comment:
i think it's small enough to be on its own. rather than expose the _obtain through the header.
If we use it more often, i'll consider.

---
In reply to: [81248196](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r81248196) [](ancestors = 81248196)